### PR TITLE
Add Sphinx documentation

### PR DESCRIPTION
**Pull request by peterhudec**
## Description of change

* Adds the [Sphinx](https://www.sphinx-doc.org/en/master/) documentation setup
* Converts the [`README.md`](https://github.com/uktrade/enquiry-mgmt-tool/blob/master/README.md) to [`README.rst`](https://github.com/uktrade/enquiry-mgmt-tool/blob/sphinx-docs/README.rst).
* Adds RST markup to docstrings

Some of the files were autogenerated by `sphinx-quickstart`.

## Test instructions

1. Follow the steps described [at the end of the new `README.rst`](https://github.com/uktrade/enquiry-mgmt-tool/blob/sphinx-docs/README.rst#compilation-to-html) to compile HTML documentation
2. You should be able to compile the documentation
3. The compiled documentation should have working links to its own sections and to external resources

### CODE REVIEW

**Review by ambrozic — APPROVED** [review]

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+.. code-block:: bash
+
+   $ npx cypress run
+
+or
+
+.. code-block:: bash
+
+   $ docker-compose run cypress run --browser firefox
```
Comment (rachaelcodes) [inline README.rst:142]: Can we change this to `docker-compose run cypress run`, as the tests are flaky on firefox?

File: README.rst [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+To set up a pre-commit hook which will prevent you from committing code with
+formatting errors, run: ``make setup-flake8-hook``
+
+Single Sign On (SSO)
+^^^^^^^^^^^^^^^^^^^^
+
+The app works out of the box with |mock-sso|_, which is part of the
+|docker-compose|_ setup. The |oauth|_ flow however only works locally when you
+set the ``AUTHBROKER_URL`` env var to ``host.docker.internal:8000``.
```
Comment (rachaelcodes) [inline README.rst:66]: I've got this running as 'http://docker.for.mac.localhost:8080' ? Maybe the host.docker.internal works, but I'm not sure about the port?

Comment (peterhudec) [reply README.rst:66]: `docker.for.mac.localhost` is actually mentioned later in the paragraph.

Comment (rachaelcodes) [reply README.rst:66]: Isn't the port used 8080, not 8000 though?

**Review by rachaelcodes — COMMENTED** [review]
👍

File: README.rst (hunk 2 above)
Comment (peterhudec) [reply README.rst:66]: Oh, I see, you meant the port, I'll fix it.

**Review by rachaelcodes — APPROVED** [review]

**Pull request merged**